File: rb/features.rb
# Realgazeta SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module RealgazetaFeatures
  def self.make_feature(name)
    case name
    when "base"
      RealgazetaBaseFeature.new
    when "test"
      RealgazetaTestFeature.new
    else
      RealgazetaBaseFeature.new
    end
  end
end

File: py/features.py
# Realgazeta SDK feature factory

from feature.base_feature import RealgazetaBaseFeature
from feature.test_feature import RealgazetaTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RealgazetaBaseFeature(),
        "test": lambda: RealgazetaTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
